File: 941-SortArrayByParity/941-SortArrayByParity.py
# Last updated: 4/29/2026, 10:26:52 AM
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        evenList = []
        oddList = []
        for num in nums:
            if num % 2 == 0:
                evenList.append(num)
            else:
                oddList.append(num)
        
        return evenList + oddList

File: 941-SortArrayByParity/941-SortArrayByParity.js
// Last updated: 4/29/2026, 10:26:57 AM
/**
 * @param {number[]} A
 * @return {number[]}
 */
// 1:21s
// O(n) time | O(n) space
var sortArrayByParity = function(A) {
    const evenArr = [];
    const oddArr = [];
    for(const num of A) {
        if(num % 2 === 0) {
            evenArr.push(num);
        } else {
            oddArr.push(num);
        }
    }
    return evenArr.concat(oddArr);
};